### PR TITLE
Added some more lighting tests, including a light baking test. Fixed the meta pass to be an actual built-in meta pass instead of a (kind off) URP pass.

### PR DESCRIPTION
**Pull request by joshua-davis**
---
### Purpose of this PR
I was adding a few tests in the lighting project, one of which is a simple light baking test. In doing so I realized the meta pass was incorrect for built-in so I fixed some changes there.

---
### Testing status
For each each of the tests, I ran them with both SG and surface shaders. Uploaded here is the diff with the tolerance set to 1 pixel (blue pixels).
| - | BuiltIn-ShaderGraph | BuiltIn-SurfaceShader | Diff |
| - | ------------- | ------------- | ----- |
| 031 |  031_Shader_GlossyEnvironmentSky | 031_Shader_GlossyEnvironmentSky | 031_Shader_GlossyEnvironmentSky diff |
| 032 | 032_Shader_GlossyEnvironmentColor | 032_Shader_GlossyEnvironmentColor | 032_Shader_GlossyEnvironmentColor diff |
| 033 | 033_Shader_HighlightsEnvironmentGradientSH | 033_Shader_HighlightsEnvironmentGradientSH | 033_Shader_HighlightsEnvironmentGradientSH diff |
| 034 | 034_Shader_HighlightsEnvironmentGradientBaked | 034_Shader_HighlightsEnvironmentGradientBaked | 034_Shader_HighlightsEnvironmentGradientBaked diff |
| 036 |  036_Lighting_Scene_DirectionalBakedDirectional | 036_Lighting_Scene_DirectionalBakedDirectional | 036_Lighting_Scene_DirectionalBakedDirectional diff |

---
### Comments to reviewers
For the meta pass, I had to actually grab the code from a surface shader and pipe the data through. Unlike with some of the other passes, the meta pass doesn't have active fields for most of the surface description fields, so I had to duplicate the pack/unpack code to built-in because of compiler errors.

Also all the "lit" shader and shader graph files are me just copying them over from the Foundation project

### CODE REVIEW

File: com.unity.shadergraph/Editor/Generation/Targets/BuiltIn/ShaderLibrary/Shim/Shims.hlsl [hunk 1]
```diff
@@ -1,7 +1,6 @@
 #ifndef UNITY_SHIMS_INCLUDED
 #define UNITY_SHIMS_INCLUDED
 
-#define BUILTIN_TARGET_API
```
Comment (joshua-davis) [inline com.unity.shadergraph/Editor/Generation/Targets/BuiltIn/ShaderLibrary/Shim/Shims.hlsl:4]: This is now defined by the pass

File: com.unity.shadergraph/Editor/Generation/Targets/BuiltIn/Editor/ShaderGraph/Targets/BuiltInLitSubTarget.cs [hunk 2]
```diff
@@ -732,7 +732,6 @@ static class LitIncludes
                 // Pre-graph
                 { CoreIncludes.CorePregraph },
                 { CoreIncludes.ShaderGraphPregraph },
-                { kMetaInput, IncludeLocation.Pregraph },
```
Comment (joshua-davis) [inline com.unity.shadergraph/Editor/Generation/Targets/BuiltIn/Editor/ShaderGraph/Targets/BuiltInLitSubTarget.cs:735]: This include caused some duplicate symbols

File: com.unity.shadergraph/Editor/Generation/Targets/BuiltIn/Editor/ShaderGraph/Includes/LightingMetaPass.hlsl [hunk 3]
```diff
@@ -1,10 +1,139 @@
 #ifndef SG_LIT_META_INCLUDED
 #define SG_LIT_META_INCLUDED
 
+#include "UnityMetaPass.cginc"
+#include "Packages/com.unity.shadergraph/Editor/Generation/Targets/BuiltIn/ShaderLibrary/SurfaceData.hlsl"
+
+SurfaceData SurfaceDescriptionToSurfaceData(SurfaceDescription surfaceDescription)
```
Comment (joshua-davis) [inline com.unity.shadergraph/Editor/Generation/Targets/BuiltIn/Editor/ShaderGraph/Includes/LightingMetaPass.hlsl:7]: These were normally shared functions for all built-in passes, but due to the surface description not containing certain fields (like metallic) in this pass I had to copy them out and remove the references to fields not in this pass.

**Review by jessebarker — APPROVED** [review]
LGTM

**Comment by jessebarker** [comment]
I'm assuming the test failures are due to the reference images (new or at least newly enabled tests)?

**Comment by joshua-davis** [comment]
> I'm assuming the test failures are due to the reference images (new or at least newly enabled tests)?

They are new yes. Hopefully once I get all the images and update it they'll all be green.

**Review by LandonTownsendUnity — APPROVED** [review]

**Pull request merged**